src/smart-components/Message/components/TextFragment/index.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import { UserMessage } from '@sendbird/chat/message';
+
+import { TOKEN_TYPES, Token } from '../../utils/tokens/types'
+import { useMessageContext } from '../../context/MessageProvider';
+import { keyGenerator } from '../../utils/tokens/keyGenerator';
+import MentionLabel from '../../../../ui/MentionLabel';
+import { USER_MENTION_PREFIX } from '../../consts';
+import LinkLabel from '../../../../ui/LinkLabel';
+import { LabelTypography } from '../../../../ui//Label';
+
+export type TextFragmentProps = {
+  tokens: Token[];
+}
+
+export default function TextFragment({
+  tokens,
+}: TextFragmentProps): React.ReactElement {
+  const messageStore = useMessageContext();
+
+  const message = messageStore?.message as UserMessage;
+  const isByMe = messageStore?.isByMe;
+  const { updatedAt, createdAt } = message;
+  return (
+    <>
+      {
+        tokens.map((token, idx) => {
+          const key = keyGenerator(createdAt, updatedAt, idx);
```

Comment:
Maybe we can use `token` as a key here. Isn't it a unique value?

Reply:
no, its not :(